src/snowflake/snowpark/dataframe.py

```diff
@@ -1334,7 +1334,19 @@ def drop(
         # An empty list of columns should be accepted as dropping nothing
         if not cols:
             raise ValueError("The input of drop() cannot be empty")
-        exprs = parse_positional_args_to_list(*cols)
+        exprs, is_variadic = parse_positional_args_to_list_variadic(*cols)
+
+        # AST.
+        stmt = self._session._ast_batch.assign()
+        if isinstance(exprs[0], str):
+            ast = stmt.expr.sp_dataframe_drop__strings
+        else:
+            ast = stmt.expr.sp_dataframe_drop__columns
+        self.set_ast_ref(ast.df)
+        set_src_position(ast.src)
```

Comment:
Consider using `with_src_position()` to eliminate one line of code from each use case.

Reply:
done